### PR TITLE
Enable modelopt quantized FLUX deployment

### PR DESCRIPTION
**Pull request by Edwardf0t1**
## Motivation

Add support for deploying NVIDIA ModelOpt FP8-quantized diffusion models in SGLang, starting with FLUX. This enables running FP8 checkpoints produced by ModelOpt's post-training quantization (`quant_algo: "FP8", quant_method: "modelopt"`) directly through the existing sglang generate / sglang serve CLI with no special flags — the pipeline auto-detects the `quantization_config` in the transformer's `config.json`.
FP8 inference uses the same CUTLASS FP8 GEMM kernels (`sgl_kernel.fp8_scaled_mm`) already used by the LLM runtime, providing faster inference on GPUs with FP8 tensor core support (Ada/Hopper/Blackwell).

## Modifications

New file: `python/sglang/multimodal_gen/runtime/layers/quantization/modelopt_fp8.py`

- ModelOptFp8Config: Quantization config that parses ModelOpt's checkpoint format (quant_algo: "FP8") and implements ignore-list matching for excluded layers (e.g., `context_embedder`, `norm_out*`, `proj_out`, `time_text_embed*`, `x_embedder`).
- ModelOptFp8LinearMethod: Linear method that creates FP8 weight + per-tensor scale parameters, processes weights after loading (transpose to column-major layout, convert scales to per-channel for CUTLASS), and calls `apply_fp8_linear` from the LLM runtime for inference.

Modified: `python/sglang/multimodal_gen/runtime/layers/quantization/__init__.py`

- Registered `"modelopt"` -> `ModelOptFp8Config` in the quantization method registry so it is auto-detected when loading a ModelOpt checkpoint.

Modified: `python/sglang/multimodal_gen/runtime/loader/fsdp_load.py`

- Added auto-dequantization in `load_model_from_full_model_state_dict:` when a checkpoint weight is `float8_e4m3fn` but the target model parameter is a higher-precision dtype (e.g., `bfloat16`), the loader automatically looks up the corresponding `weight_scale` and dequantizes (`float_value = fp8_value * weight_scale`). This handles modules like diffusers' `AdaLayerNormZero` that contain linear layers quantized in the ModelOpt checkpoint but are not FP8-aware in the model definition (they use standard `nn.Linear` rather than SGLang's quantization-aware `LinearBase`).

## Accuracy Tests

Example usage:
```
sglang generate   --model-path /home/omniml_data_3/zhiyuc/checkpoints/FLUX.1-dev-FP8   --prompt "A beautiful sunset over the mountains"   --save-output --output-path outputs
```

- [x] Verify ModelOptFp8Config correctly parses the quantization_config from a ModelOpt FP8 checkpoint
- [x] Verify ignore-list matching correctly identifies excluded layers vs quantized layers
- [x] Generate an image with the FP8 FLUX checkpoint and confirm visual quality

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/loader/fsdp_load.py [hunk 1]
```diff
@@ -285,6 +285,25 @@ def load_model_from_full_model_state_dict(
         else:
             target_dtype = meta_sharded_param.dtype
 
+        # Auto-dequantize: checkpoint has FP8 weight but the model parameter
+        # is a higher-precision type (e.g. modules that don't accept
+        # quant_config like diffusers' AdaLayerNormZero).  Multiply by the
+        # per-tensor weight_scale so the loaded value matches the original
+        # unquantized weight.
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/loader/fsdp_load.py]: move the logic to a helper in this file

**Review by mickqian — COMMENTED** [review]
could you attach reproducible command and output?

**Review by mickqian — COMMENTED** [review]
sorry for the late review, appreciate this

File: python/sglang/multimodal_gen/runtime/loader/fsdp_load.py (hunk 1 above)
Comment (Edwardf0t1) [reply python/sglang/multimodal_gen/runtime/loader/fsdp_load.py]: Created a helper function `_maybe_dequantize_fp8()`

**Comment by Edwardf0t1** [comment]
> could you attach reproducible command and output?

The command is included in the PR description: 

`sglang generate   --model-path /home/omniml_data_3/zhiyuc/checkpoints/FLUX.1-dev-FP8   --prompt "A beautiful sunset over the mountains"   --save-output --output-path outputs`

The output is attached.
A_beautiful_sunset_over_the_mountains_20260308-064001_dda7dec1

**Comment by Edwardf0t1** [comment]
> sorry for the late review, appreciate this

@mickqian Thanks for the review, addressed your comments, PTAL.

**Review by mickqian — APPROVED** [review]
